### DISCUSSION TITLE
Proxy support?

### DISCUSSION THREAD

**Discussion by ralph240574** [googleapis/nodejs-pubsub#1321] [Q&A]
I am trying to use the library with a proxy server. Is there any support for this?

**Comment by feywind** [comment]
Hi @ralph240574 , our grpc-js person suggests this:

https://github.com/grpc/grpc/blob/master/doc/environment_variables.md

The first two lines in that doc are describing proxy-related environment variables that are still supported in grpc-js (which we use now).

Hope that helps!

**Reply by ralph240574** [reply]
Thanks, but this document refers to the gRPC C core-based implementations. I don't think they are used with the pubsub client.

**Reply by feywind** [reply]
That doc is for the C implementation, yeah. I'm told that the proxy lines do still work in the JavaScript version that we do still use:

```
grpc_proxy, https_proxy, http_proxy The URI of the proxy to use for HTTP CONNECT support. These variables are checked in order, and the first one that has a value is used.

no_grpc_proxy, no_proxy A comma separated list of hostnames to connect to without using a proxy even if a proxy is set. These variables are checked in order, and the first one that has a value is used.
```
